tests/monotouch-test/CoreFoundation/MutableString.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using Foundation;
+using CoreFoundation;
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.CoreFoundation {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class MutableString {
+
+		[Test]
+		public void CreateString0 ()
+		{
+			using (var s = new CFMutableString ()) {
+				Assert.That (s.Handle, Is.Not.EqualTo (IntPtr.Zero), "Handle");
+				Assert.That (s.ToString (), Is.Empty, "ToString");
+			}
+		}
+
+		[Test]
+		public void CreateString1 ()
+		{
+			Assert.Throws<ArgumentNullException> (() => new CFMutableString ((string)null), "null");
+			using (var s = new CFMutableString ("bonjour!")) {
```

Comment:
More complex test strings would be nice too: emojis, bidi characters, unicode characters with multiple code points, etc.

Reply:
Tests are to validate the bindings, not the API. That historically caused a lot of troubles (false positives) when run on devices with different OS versions or installed/default languages.

Is there something specific you think would cause problems ?

Reply:
Yes, I'd like to confirm if Mono's understanding of Unicode is the same as Apple's, to ensure that a roundtrip string -> CFString -> string doesn't end up with something corrupted.

For instance: `string.Length` is "the number of Char objects in this instance, not the number of Unicode characters.", while `CFStringAppendCharacters` takes `numChars:
The number of Unicode characters in chars`. I don't trust documentation, but it seems there is a possibility that `CFStringAppendCharacters` can go off into random memory looking for more characters than we actually provided.

Unicode is hard 😒